### PR TITLE
Fix dox

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

**Comment by sdroege** [comment]
:+1:

**Comment by GuillaumeGomez** [comment]
@sdroege: Two tests are failing, any idea what's happening?

File: src/source.rs [hunk 1]
```diff
@@ -1,11 +1,13 @@
-// Copyright 2013-2015, The Gtk-rs Project Developers.
+// Copyright 2013-2018, The Gtk-rs Project Developers.
 // See the COPYRIGHT file at the top-level directory of this distribution.
 // Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
 
 use std::cell::RefCell;
 use std::mem::transmute;
-#[cfg(any(all(feature = "v2_36", unix), feature = "dox"))]
+#[cfg(any(feature = "v2_36", unix))]
 use std::os::unix::io::RawFd;
```
Comment (EPashkin) [inline src/source.rs]: This is produce unused warning on travis linux 3.4

File: .travis.yml [hunk 2]
```diff
@@ -1,16 +1,53 @@
...
     packages:
     - libgtk-3-dev
+before_install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew unlink python; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install gtk+3 cairo atk; fi
 script:
+  - cargo doc --features "dox"
   - cargo test
   - rustc --version
```
Comment (EPashkin) [inline .travis.yml]: Maybe this line better removed as it already printed after rust installation
or moved as first line

**Comment by sdroege** [comment]
Also `test_paths` is failing for some reason, maybe because of travis using symbolic links and one of the parts resolving them?

```
---- translate::tests::test_paths stdout ----
	thread 'translate::tests::test_paths' panicked at 'assertion failed: `(left == right)`
  left: `Some("/private/var/folders/bb/n7t3rs157850byt_jfdcq9k80000gn/T/glib-test.7UEbO9t321jY")`,
 right: `Some("/var/folders/bb/n7t3rs157850byt_jfdcq9k80000gn/T/glib-test.7UEbO9t321jY")`', src/translate.rs:1521:9
note: Run with `RUST_BACKTRACE=1` for a backtrace.
failures:
    translate::tests::test_paths
```

**Comment by sdroege** [comment]
And also build failing due to missing `xcb-shm`:

```
+cargo build --features gtk_3_18 --jobs 1
[...]
error: failed to run custom build command for `cairo-sys-rs v0.6.0 (https://github.com/gtk-rs/cairo#7b52ef54)`
process didn't exit successfully: `/Users/travis/build/gtk-rs/glib/_examples/target/debug/build/cairo-sys-rs-5454e01fbd6b4aa1/build-script-build` (exit code: 1)
--- stderr
`"pkg-config" "--libs" "--cflags" "cairo >= 1.10"` did not exit successfully: exit code: 1
--- stderr
Package xcb-shm was not found in the pkg-config search path.
Perhaps you should add the directory containing `xcb-shm.pc'
to the PKG_CONFIG_PATH environment variable
Package 'xcb-shm', required by 'cairo', not found
```

So many things at once!

**Comment by sdroege** [comment]
The main context thing is probably fixed by https://github.com/gtk-rs/glib/pull/316

**Comment by sdroege** [comment]
Someone kick CI, everything but the xcb-shm problem should be gone now

**Comment by GuillaumeGomez** [comment]
I restarted one already. :)

File: .travis.yml [hunk 3]
```diff
@@ -1,16 +1,54 @@
...
+sudo: true
 addons:
   apt:
     packages:
     - libgtk-3-dev
+before_install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew unlink python; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install gtk+3 cairo atk; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then export PKG_CONFIG_PATH=/usr/local/lib/pkgconfig:/opt/X11/lib/pkgconfig; fi
```
Comment (EPashkin) [inline .travis.yml]: Separate ifs with same condition is special?

Comment (GuillaumeGomez) [reply .travis.yml]: Just more readable (from my point of view).

**Comment by EPashkin** [comment]
@GuillaumeGomez Its good.
If you don't plan squash commits, please rename "don't test 3.18" to "don't test 3.18 on osx"

**Comment by GuillaumeGomez** [comment]
I'll squash/rename.

**Comment by GuillaumeGomez** [comment]
And done.

File: .travis.yml (hunk 3 above)
Comment (EPashkin) [reply .travis.yml]: IMHO multi-line commands is allowed.
Currently only found not full multi-command example in https://github.com/gtk-rs/gio/blob/master/.travis.yml#L15-L17,
but I seen real multiple commands somewhere.

**Pull request merged**